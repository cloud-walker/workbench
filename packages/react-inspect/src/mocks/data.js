export const und = undefined
export const nil = null
export const num = 666
export const str = 'cawabongaaa!'
export const fun = a => {
  const b = 2

  while (a < 10) {
    a++
  }

  return a + b
}
const obj = {und, nil, num, str, fun}
export const arr = Object.values(obj)

obj.arr = arr
obj.obj = Object.assign({}, obj)

export default obj

const Circ = function() {
  this.foo = 'bar'
  this.circ = this
}

export const circ = new Circ()
