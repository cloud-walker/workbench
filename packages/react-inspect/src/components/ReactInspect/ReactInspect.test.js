import React from 'react'
import {shallow} from 'enzyme'

import Component from './index'
import data, {circ} from '../../mocks/data'

const subject = shallow(<Component data={data} />)

/* eslint-env jest */
describe(`${Component.displayName} component`, () => {
  it('should render properly', () => {
    expect(subject).toMatchSnapshot()
  })

  it('should throw an error if the data is circular', () => {
    expect(() => subject.setProps({data: circ})).toThrowError()
  })
})
