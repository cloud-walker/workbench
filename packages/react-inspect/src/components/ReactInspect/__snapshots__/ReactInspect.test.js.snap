// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactInspect component should render properly 1`] = `
<ReactInspectLayout>
  <ReactInspectDataHandler
    data={
      Object {
        "arr": Array [
          undefined,
          null,
          666,
          "cawabongaaa!",
          [Function],
        ],
        "fun": [Function],
        "nil": null,
        "num": 666,
        "obj": Object {
          "arr": Array [
            undefined,
            null,
            666,
            "cawabongaaa!",
            [Function],
          ],
          "fun": [Function],
          "nil": null,
          "num": 666,
          "str": "cawabongaaa!",
          "und": undefined,
        },
        "str": "cawabongaaa!",
        "und": undefined,
      }
    }
    outer={true}
  />
</ReactInspectLayout>
`;
